src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -138,6 +138,25 @@ public async Task SetupAsync()
 
             _expectingSettingsAck = true;
 
+
+            // Receive the initial SETTINGS frame from the peer.
+            try
+            {
+                FrameHeader frameHeader = await ReadFrameAsync(initialFrame: true).ConfigureAwait(false);
+                if (frameHeader.Type != FrameType.Settings || frameHeader.AckFlag)
+                {
+                    throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
+                }
+
+                // Process the initial SETTINGS frame. This will send an ACK.
+                ProcessSettingsFrame(frameHeader);
+            }
+            catch
+            {
+                // ISSUE 31315: Determine if/how to expose HTTP2 error codes
+                throw new HttpRequestException(SR.net_http_invalid_response);
```

Comment:
We typically would insert the the exception from the 'catch' into the HttpRequestException.  Otherwise, it is difficult to diagnose anything. So, this should be changed to something like:

```c#
catch (Exception ex)
{
    throw new HttpRequestException(SR.net_http_invalid_response, ex);
```

Reply:
updated.